doc/osh-help.md

```diff
@@ -805,6 +806,135 @@ these are discouraged.
 
 <!--    -R VAR     True if the variable VAR has been set and is a nameref variable. -->
 
+<a h4="printf">printf</h4>
+
+    printf FLAG? FMT ARGS*
+
+Displays text according to a format string.
+
+Flags:
+    -v VAR  Write output in variable VAR instead of standard output.
+
+FMT:
+    String that can contain three types of objects: 1) characters that will be
+    printed, 2) escape sequences that are interpreted and then printed,
+    3) format specifiers that print the strings passed to printf as ARGS, but
+    after giving it a determinate format (if not enough ARGS are passed, zero
+    or a nul string will be used)
+
+Escape sequences:
```

Comment:
I think it would make sense to create a new help topic for the character escapes.  So here is no duplication between echo and printf.

I can do that (because one surprising difference/bug is that `echo`   in bash uses  `\0ddd` while printf uses `\ddd`  !)

If you look at `_C_STRING_COMMON`  in `frontend/lexer_def.py`, it has the parts they share, and the parts they don't share are in `PrintfOuter` and `ECHO_E_DEF`